ibis/impala/compiler.py

```diff
@@ -95,54 +95,59 @@ def compile(self):
 
         # If any subqueries, translate them and add to beginning of query as
         # part of the WITH section
-        with_frag = self.format_subqueries()
+        with_frag = self.format_subqueries(params=params)
 
         # SELECT
-        select_frag = self.format_select_set()
+        select_frag = self.format_select_set(params=params)
 
         # FROM, JOIN, UNION
-        from_frag = self.format_table_set()
+        from_frag = self.format_table_set(params=params)
 
         # WHERE
-        where_frag = self.format_where()
+        where_frag = self.format_where(params=params)
 
         # GROUP BY and HAVING
-        groupby_frag = self.format_group_by()
+        groupby_frag = self.format_group_by(params=params)
 
         # ORDER BY and LIMIT
-        order_frag = self.format_postamble()
+        order_frag = self.format_postamble(params=params)
```

Comment:
This seems like a lot of parameter "threading". Any reason to have this as an argument versus a `self.params` when the query is being compiled?